packages/grafana-ui/src/components/uPlot/plugins/TooltipPlugin2.tsx

```diff
@@ -190,7 +205,15 @@ export const TooltipPlugin2 = ({ config, render }: TooltipPlugin2Props) => {
 
     // fires on data value hovers/unhovers (before setSeries)
     config.addHook('setLegend', (u) => {
-      let _isHoveringNow = _plot!.cursor.idxs!.some((v) => v != null);
+      let hoveredSeriesIdx = _plot!.cursor.idxs!.findIndex((v, i) => i > 0 && v != null);
+      let _isHoveringNow = hoveredSeriesIdx !== -1;
+
+      // in mode: 2 uPlot won't fire the proximity-based setSeries (below)
+      // so we set closestSeriesIdx here instead
+      // TODO: setSeries only fires for TimeSeries & Trend...not state timeline or statsus history
+      if (hoverMode === TooltipHoverMode.xyOne) {
+        closestSeriesIdx = hoveredSeriesIdx;
```

Comment:
this _should_ remove the need to do this downstream in the downstream components for StatusHistory and XYChart and Heatmap.